PROTOCOL.md

```diff
@@ -0,0 +1,75 @@
+
+### Invitations
+
+A conversation is initiated by a client sending an [Invitation](https://github.com/xmtp/proto/blob/main/proto/message_contents/invitation.proto) to one or more participants. The invitation contains a randomly generated name of the topic to be used for the conversation messages, and randomly generated secret key material to be used for message encryption. It also indicates the encryption algorithm to use for message encryption.
+
+Invitations are published into invite topics. Each blockchain account is automatically assigned an invite topic named `invite-<account address>`. It contains all the invitations both sent and received by the account owner. It allows the client to reconstruct the list of all past conversations and to find their corresponding message topics.
+
+Invitations are encrypted using a key derived from the pre-keys of the invitation sender and recipient. Consequently the sender and the recipient of an invitation are present in the header, outside of the encrypted payload, and visible to outside observers (i.e. nodes).
+
+### Messages
+
+[Messages](https://github.com/xmtp/proto/blob/main/proto/message_contents/xmtp_envelope.proto) are published into conversation topics indicated by the corresponding invitation. Conversation topic names follow the format `m-<base64 encoded random 32bytes>`.
+
+Message content is encoded using a content type framework represented by the [EncodedContent type](https://github.com/xmtp/proto/blob/main/proto/message_contents/xmtp_envelope.proto). Usage of the content type framework is governed by [XIP-5](https://github.com/xmtp/XIPs/blob/main/XIPs/xip-5-message-content-types.md).
+
+The bytes of the `EncodedContent` are wrapped in [SignedContent](https://github.com/xmtp/proto/blob/main/proto/message_contents/xmtp_envelope.proto), signed using the pre-key of the sender. The signature covers the bytes of the `EncodedContent` and the bytes of the [MessageHeaderV2](https://github.com/xmtp/proto/blob/main/proto/message_contents/xmtp_envelope.proto) to frustrate replay attacks. `SignedContent` is then encrypted using the key material from the invitation and wrapped in [Ciphertext](https://github.com/xmtp/proto/blob/main/proto/message_contents/ciphertext.proto). Finally [MessageV2](https://github.com/xmtp/proto/blob/main/proto/message_contents/xmtp_envelope.proto) combines the `Ciphertext` and the bytes of the message header.
+
+## V1
+
+Previous version of the protocol (further V1) had the differences described below.
+
+The networks support both versions of the protocol in order to allow retrieving historic conversations correctly. V1 conversations can still receive new messages, if the parties initiated them before V2 protocol rollout.
```

Comment:
```suggestion
The XMTP network supports both versions of the protocol to allow for correct retrieval of historical conversations. V1 conversations can still receive new messages if the parties initiated them before the V2 protocol rollout.
```

Reply:
Not sure how you feel about this, Martin - but since we mention V2 here - do we need to acknowledge somewhere that all of the content above the V1 section heading describes XMTP V2?

Reply:
I agree, we should. But I'm struggling with how. Ideally it would be right at the top of the document, but it doesn't fit well with the way the Intro is written.  Do you think it could go above the Intro, just say something like.

> Current version of the protocol is V2. This is what this document focuses on. Prior versions are described at the end of the document.

Reply:
I think right above the "Intro" is a great place for this information, @mkobetic! Perhaps exactly what you wrote, with just a few tweaks - something like this?

>This document focuses on describing the current version of the Extensible Message Transport Protocol: XMTP V2.
>
>To learn about the previous version of the protocol, see [V1](#V1).

Reply:
Love it!